error.go

```diff
@@ -8,11 +8,18 @@ import (
 
 // APIError provides error information returned by the OpenAI API.
 type APIError struct {
-	Code           any     `json:"code,omitempty"`
-	Message        string  `json:"message"`
-	Param          *string `json:"param,omitempty"`
-	Type           string  `json:"type"`
-	HTTPStatusCode int     `json:"-"`
+	Code           any        `json:"code,omitempty"`
+	Message        string     `json:"message"`
+	Param          *string    `json:"param,omitempty"`
+	Type           string     `json:"type"`
+	HTTPStatusCode int        `json:"-"`
+	InnerError     InnerError `json:"innererror,omitempty"`
+}
+
+// InnerError Azure Content filtering.
```

Comment:
@ZeroDeng01  Is this the official documentation for 'innererror'? It seems there are various other fields as well. The fields seem to vary depending on the API.
https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#examples

Reply:
@vvatanabe At present, there seems to be no official document about InnerError in Azure Openai, and I have tested that adding InnerError in Azure Openai is mainly to increase the error information of content review, and there are no more new InnerError data structures for other Azure Openai-related errors. So I am not sure whether Azure will unify the data structure of this error message in the future. I thought about using any data type, but in order to be compatible with too many unknown data structure forms, I am not sure if there will be other problems. In summary, I chose to define an InnerError structure applicable to official Azure Openai, but It is not guaranteed whether it is compatible with other APIs of Azure, because go-openai mainly works for openai.

Reply:
@ZeroDeng01 It looks like innererror field is also supported by azure-sdk-for-go.

- https://github.com/Azure/azure-sdk-for-go/blob/49b36611277b492bacf332bd765cafdb0f27f942/sdk/ai/azopenai/custom_models.go#L81-L87
- https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/ai/azopenai/models.go#L480C1-L500